### PR TITLE
Renaming files, classes, and interfaces for improved clarity about their purpose

### PR DESCRIPTION
**Pull request by tiagobento**
As time passed, some important classes remained with ambiguous names. Now that the complexity of the codebase has increased, I'm refreshing the names of some files to better reflect their purpose.

### CODE REVIEW

File: packages/notifications/README.md [hunk 1]
```diff
@@ -15,7 +15,7 @@ The library is separated into two submodules:
 
   to use the core:
 
-  - `import { NotificationsApi } from "@kie-tools-core/notifications/dist/api"`
+  - `import { NotificationsChanelApi } from "@kie-tools-core/notifications/dist/api"`
```
Comment (caponetto) [inline packages/notifications/README.md]:
```suggestion
  - `import { NotificationsChannelApi } from "@kie-tools-core/notifications/dist/api"`
```
Chanel -> Channel

File: packages/notifications/README.md [hunk 2]
```diff
@@ -27,14 +27,14 @@ The library is separated into two submodules:
   to use the vscode classes:
 
   ```ts
-  import { VsCodeNotificationsApi } from "@kie-tools-core/i18n/dist/react-components";
+  import { VsCodeNotificationsChannelApiImpl } from "@kie-tools-core/vscode-extension/dist/vscode";
 
-  const api: NotificationsApi = new VsCodeNotificationsApi(workspaceApi, i18n);
+  const api: NotificationsChanelApi = new VsCodeNotificationsChannelApiImpl(workspaceApi, i18n);
```
Comment (caponetto) [inline packages/notifications/README.md]:
```suggestion
  const api: NotificationsChannelApi = new VsCodeNotificationsChannelApiImpl(workspaceApi, i18n);
```
Chanel -> Channel

**Review by caponetto — COMMENTED** [review]
Found two typos. Other than this, looks great to me! Thanks, @tiagobento!

File: packages/notifications/README.md (hunk 2 above)
Comment (tiagobento) [reply packages/notifications/README.md]: Done in [16fd737](https://github.com/kiegroup/kie-tools/pull/1122/commits/16fd7376d29962867b4e4da2bba02daeefc842c6)

**Review by caponetto — APPROVED** [review]

**Pull request merged**